codecoverage.py

```diff
@@ -19,6 +19,12 @@
 
 
 TEST_PLATFORMS = ['test-linux64-ccov/debug', 'test-windows10-64-ccov/debug']
+ALL_STATUSES = ['completed', 'failed', 'exception', 'unscheduled', 'pending', 'running']
```

Comment:
If the status is one of `unscheduled`, `pending` or `running`, we should wait for them to finish.

Reply:
Maybe define a `FINISHED_STATUSES` and `IN_PROGRESS_STATUSES`, the first with `completed`, `failed` and `exception` and the second with the other three.
Then in `download_coverage_artifacts`, right before the loop which chooses the tasks, add another loop which waits until all the tasks's statuses are in FINISHED_STATUSES.